source/walkthroughs/wt_active_learning.rst

```diff
@@ -0,0 +1,227 @@
+Step-by-Step Guide to Active Learning with Log Data in MindMeld
+===============================================================
+
+Active Learning is an approach to continuously and iteratively pick data that is most informative for the models to train on from a pool of unlabelled data-points. By picking such data-points, active learning provides a higher chance of improving the accuracy of the model with fewer training examples. It also greatly reduces the number of queries to be annotated by humans. MindMeld provides this inbuilt functionality to select these must-have queries from existing data or additional logs and datasets to get the best out of your conversational applications.
+
+The following step-by-step guide will use the :doc:`HR Assistant <../blueprints/hr_assistant>` blueprint for showcasing MindMeld's active learning functionality and the different customizations involved.
+
+Step 1: Setup a MindMeld App with Log Resources
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The first step for running active learning is to set up your MindMeld app with log data configured for the app. This log data can either be in the form of labelled text files (similar to the train and test files in the app), or an unlabelled text file containing raw text queries across all domains and intents.
+
+For the purpose of this tutorial, we will generate 'logs' for the current HR Assistant blueprint using the MindMeld :doc:`Data Augmentation <../userguide/augmentation>` pipeline. These paraphrases will serve as additional queries for the app to train on. After we have figured out the best hyperparameters using the tuning step, we'll select the best qureries from the data augmentation logs (files with the pattern ``train-augmented.txt``). Adding these queries to the train files of the assistant can improve performance of the classifiers.
+
+Step 2: Define Active Learning Config
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This section shows the customized active learning configuration setting in ``config.py`` for the HR Assistant app. A detailed description for each of the components can be found :ref:`here <al_config>`.
+
+.. code-block:: python
+
+    ACTIVE_LEARNING_CONFIG = {
+        "output_folder": "hr_assistant_active_learning",
+        "pre_tuning": {
+            "train_pattern": ".*train.txt",
```

Comment:
Should the regex allow text after "train" and escape the period? ".*train.*\.txt"

Reply:
Generally, yes! The default configuration allows for text after train and escapes the period. The config mentioned in this file is specific to the blueprint app, and this regex helps differentiate between train and train-augmented files.